### ISSUE TITLE
REST Mapping for file-systems & valid-value-sets  invalid

### ISSUE THREAD

**Issue by planetf1** [odpi/egeria#6010]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

On launching the chassis with latest dependencies (including spring updates)
```
***************************
APPLICATION FAILED TO START
***************************

Description:

Invalid mapping pattern detected: /file-systems/{fileSystemGUID}/folders/{fileSystemGUID}/attach
                                       ^
Not allowed to capture 'fileSystemGUID' twice in the same pattern

Action:

Fix this pattern in your application or switch to the legacy parser implementation with 'spring.mvc.pathpattern.matching-strategy=ant_path_matcher'.
```

### Expected Behavior

server chassis starts normally

### Steps To Reproduce

Launch chassis with latest dependabot updates

### Environment

```markdown
- Egeria:
- OS:
- Java:
- Browser (for UI issues):
- Additional connectors and integration:
```

### Any Further Information?

See https://github.com/odpi/egeria/pull/6007

**Comment by planetf1** [comment]
The code in FileSystemOnboardingResource.java fails this validation in the methods:
```
    /**
     * Links a folder to a file system. The folder is not changed.
     *
     * @param serverName name of calling server
     * @param userId calling user
     * @param fileSystemGUID unique identifier of the file system in the catalog
     * @param folderGUID unique identifier of the folder in the catalog
     * @param requestBody dummy request body
     *
     * @return void or
     * InvalidParameterException one of the parameters is null or invalid or
     * PropertyServerException problem accessing property server or
     * UserNotAuthorizedException security access problem.
     */
    @PostMapping(path = "/file-systems/{fileSystemGUID}/folders/{fileSystemGUID}/attach")

    public VoidResponse attachFolderToFileSystem(@PathVariable                  String          serverName,
                                                 @PathVariable                  String          userId,
                                                 @PathVariable                  String          fileSystemGUID,
                                                 @PathVariable                  String          folderGUID,
                                                 @RequestBody(required = false) NullRequestBody requestBody)
    {
        return restAPI.attachFolderToFileSystem(serverName, userId, fileSystemGUID, folderGUID, requestBody);
    }


    /**
     * Removed the link between a folder and a file system.
     *
     * @param serverName name of calling server
     * @param userId calling user
     * @param fileSystemGUID unique identifier of the file system in the catalog
     * @param folderGUID unique identifier of the folder in the catalog
     * @param requestBody dummy request body
     *
     * @return void or
     * InvalidParameterException one of the parameters is null or invalid or
     * PropertyServerException problem accessing property server or
     * UserNotAuthorizedException security access problem.
     */
    @PostMapping(path = "/file-systems/{fileSystemGUID}/folders/{fileSystemGUID}/detach")

    public VoidResponse detachFolderFromFileSystem(@PathVariable                  String          serverName,
                                                   @PathVariable                  String          userId,
                                                   @PathVariable                  String          fileSystemGUID,
                                                   @PathVariable                  String          folderGUID,
                                                   @RequestBody(required = false) NullRequestBody requestBody)
    {
        return restAPI.detachFolderFromFileSystem(serverName, userId, fileSystemGUID, folderGUID, requestBody);
    }
```

**Comment by planetf1** [comment]
Even with the legacy matcher the same error is reported
```
java -Dspring.mvc.pathpattern.matching-strategy=ant_path_matcher -jar ~/src/egeria/open-metadata-implementation/server-chassis/server-chassis-spring/target/server-chassis-spring-${REL}.jar
```

**Comment by planetf1** [comment]
Looking at the actual API, it appears this may in fact be an error - with the latter fileSystemGUID actually intended to be folderGUID

**Comment by planetf1** [comment]
This is confirmed by IntelliJ which warns that folderGUID has no mapping ie
<img width="1196" alt="Screenshot 2021-11-30 at 16 41 53" src="https://user-images.githubusercontent.com/7292002/144089703-d34419fc-8d8c-4e17-8814-394a2501b4d0.png">

**Comment by planetf1** [comment]
Fixed (tbc) in 858e6d8be4114c371273f37e07afe09a03e7c6b1

**Comment by planetf1** [comment]
Additional validation failure:
```
Invalid mapping pattern detected: /valid-value-sets/by-name}
[INFO]                          ^
[INFO] Missing preceding open capture character before variable name{
```

This is caused by a rogue bracket:
```
    @PostMapping(path = "/valid-value-sets/by-name}")

    public ValidValueSetsResponse getValidValueSetByName(@PathVariable String          serverName,
                                                         @PathVariable String          userId,
                                                         @RequestBody  NameRequestBody requestBody,
                                                         @RequestParam int             startFrom,
                                                         @RequestParam int             pageSize)

 ```
 The intent of this URL does not call for a {variable} so it looks like the } should be delete

**Comment by planetf1** [comment]
Fixed in e001ded9adb4351c249ba154f8946adf6712371b
